crates/transaction-emitter/src/lib.rs

```diff
@@ -203,40 +206,57 @@ struct SubmissionWorker {
     rng: ::rand::rngs::StdRng,
 }
 
+// Note, there is an edge case that can occur if the transaction emitter
+// bursts the target node too fast, and the emitter doesn't handle it
+// very well, instead waiting up until the timeout for the target seqnum
+// to progress, even though it never will. See more here:
+// https://github.com/aptos-labs/aptos-core/issues/1565
 impl SubmissionWorker {
     #[allow(clippy::collapsible_if)]
     async fn run(mut self, gas_price: u64) -> Vec<LocalAccount> {
         let wait_duration = Duration::from_millis(self.params.wait_millis);
+        let total_start_time = Instant::now();
+        let mut total_num_requests = 0;
         while !self.stop.load(Ordering::Relaxed) {
             let requests = self.gen_requests(gas_price);
             let num_requests = requests.len();
-            let start_time = Instant::now();
-            let wait_until = start_time + wait_duration;
+            total_num_requests += num_requests;
+            let loop_start_time = Instant::now();
+            let wait_until = loop_start_time + wait_duration;
             let mut txn_offset_time = 0u64;
             for request in requests {
                 let cur_time = Instant::now();
-                txn_offset_time += (cur_time - start_time).as_millis() as u64;
+                txn_offset_time += (cur_time - loop_start_time).as_millis() as u64;
                 self.stats.submitted.fetch_add(1, Ordering::Relaxed);
                 let resp = self.client.submit(&request).await;
                 if let Err(e) = resp {
                     warn!("[{:?}] Failed to submit request: {:?}", self.client, e);
                 }
             }
             if self.params.wait_committed {
-                self.update_stats(start_time, txn_offset_time, num_requests)
+                self.update_stats(loop_start_time, txn_offset_time, num_requests)
                     .await
             }
             let now = Instant::now();
             if wait_until > now {
                 time::sleep(wait_until - now).await;
             }
         }
+        if self.params.check_stats_at_end {
```

Comment:
I am wondering if this should be a flag or why not just make this behavior default? Do you see any issue with that? More config knobs just make this tool very hard to use.

Reply:
I don't want to impose on any other potential users but if you think this is a sensible default, I can make this an opt out flag instead.

Reply:
I think this makes sense as default behavior - I am okay with an opt out flag but no flag is okay as well IMO :)